### PR TITLE
feat(Anime genzai): add presence

### PR DESCRIPTION
**Pull request by shiseni**
## Description 
Add presence for [Anime genzai](https://animegenzai.net/)
* Resolves: #5249

Settings:
1. Multilanguage,
2. Privacy,
3. Show covers,
4. Show timestamps,
5. Show buttons.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

| Home | Viewing anime| Viewing TV Show | Searching anime|
| --- | --- | --- | --- |
| image | image | image | image |

</details>

### CODE REVIEW

File: websites/A/Anime Genzai/metadata.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.7",
+  "service": "Anime Genzai",
+  "author": {
+    "id": "257151906038677504",
+    "name": "SHISEN#8023"
```
Comment (theusaf) [inline websites/A/Anime Genzai/metadata.json:6]:
```suggestion
    "name": "SHISEN"
```

File: websites/A/Anime Genzai/metadata.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  "color": "#f5c518",
+  "tags": [
+    "animegenzai",
+    "anime"
+  ],
+	"iframe": true,
+  "category": "anime",
+  "settings": [
+		{
+			"id": "lang",
```
Comment (theusaf) [inline websites/A/Anime Genzai/metadata.json:24]: This file is not formatted correctly. Use `yarn format` to format this file.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/A/Anime Genzai/metadata.json (hunk 2 above)
Comment (shiseni) [reply websites/A/Anime Genzai/metadata.json:24]: Oh... I forgot about it.

File: websites/A/Anime Genzai/presence.ts [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+			searchFor: "general.searchFor",
+			play: "general.watchingVid",
+			pause: "general.paused",
+			searchSomething: "general.searchSomething",
+			watchingMovie: "general.watchingMovie",
+			watchVideoButton: "general.buttonWatchVideo",
+			buttonViewPage: "general.buttonViewPage",
+			viewPage: "general.viewPage",
+			viewMovie: "general.viewMovie",
+			chapter: "general.chapter",
```
Comment (DarkVillager) [inline websites/A/Anime Genzai/presence.ts:22]: Its unused
```suggestion
```

File: websites/A/Anime Genzai/presence.ts [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+	Play = "https://i.imgur.com/juDzimk.png",
+	Pause = "https://i.imgur.com/GbAMPHI.png",
+}
+
+let video = {
+		duration: 0,
+		currentTime: 0,
+		paused: true,
+	},
+	strings: Awaited<ReturnType<typeof getStrings>>;
```
Comment (DarkVillager) [inline websites/A/Anime Genzai/presence.ts:44]: This is a better way to do this (1/3):
```suggestion
strings: Awaited<ReturnType<typeof getStrings>>,
	oldLang: string = null;
```

File: websites/A/Anime Genzai/presence.ts [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+		},
+		[privacy, logo, time, buttons] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
```
Comment (DarkVillager) [inline websites/A/Anime Genzai/presence.ts:69]: This is a better way to do this (2/3):
```suggestion
		[newLang, privacy, logo, time, buttons] = await Promise.all([
			presence.getSetting<string>("lang").catch(() => "es"),
			presence.getSetting<boolean>("privacy"),
```

File: websites/A/Anime Genzai/presence.ts [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+			presence.getSetting<boolean>("logo"),
+			presence.getSetting<boolean>("time"),
+			presence.getSetting<boolean>("buttons"),
+		]),
+		{ pathname, href } = document.location,
+		path = pathname.split("/")[1];
+
+	if (!strings) strings = await getStrings();
```
Comment (DarkVillager) [inline websites/A/Anime Genzai/presence.ts:77]: This is a better way to do this (3/3):
```suggestion
	if (oldLang !== newLang || !strings) {
		oldLang = newLang;
		strings = await getStrings();
	}
```

File: websites/A/Anime Genzai/presence.ts [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+		if (video.paused || !time) {
+			delete presenceData.startTimestamp;
+			delete presenceData.endTimestamp;
+		} else {
+			[presenceData.startTimestamp, presenceData.endTimestamp] =
+				presence.getTimestamps(video.currentTime, video.duration);
+		}
+	}
+
+	if (!logo || privacy) presenceData.largeImageKey = Assets.Logo;
```
Comment (DarkVillager) [inline websites/A/Anime Genzai/presence.ts:221]: You could add a check to see if it has even changed (Optional tho):
```suggestion
	if ((privacy || !logo) && presenceData.largeImageKey !== Assets.Logo) presenceData.largeImageKey = Assets.Logo;
```

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**